### PR TITLE
[Dotnet] Starts a single server instance for integration tests.

### PR DESCRIPTION
**Pull request by batiati**
Changes the integration tests logic to share the same server instance for running all tests instead of a fresh instance per test.

It also removes the `[DoNotParallelize]` attribute.

Before
```
Passed!  - Failed:     0, Passed:    72, Skipped:     0, Total:    72, Duration: 2 m 45 s
```

After
```
Passed!  - Failed:     0, Passed:    72, Skipped:     0, Total:    72, Duration: 1 m 28 s
```

### CODE REVIEW

**Comment by matklad** [comment]
🥳

File: src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -142,94 +163,98 @@ public void ConstructorAndFinalizer()
...
     }
 
     [TestMethod]
-    [DoNotParallelize]
     public void CreateTransfers()
     {
-        using var server = new TBServer();
-        using var client = GetClient(server.Address);
+        Assert.IsNotNull(server);
+        Assert.IsNotNull(client);
```
Comment (matklad) [inline src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs]: Tbh, I’d drop these extra asserts from tests: this is something forced upon us by the testing framework, but we are using an idiomatic testing framework, and so can accept slight degradation of safety (in tests !!), so as not to swim upstream against C# here.

File: src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs [hunk 2]
```diff
@@ -1540,38 +1563,38 @@ private void ConcurrentTasksDispose(bool isAsync)
...
         };
 
-        _ = client.CreateTransfers(new Transfer[] { transfer });
+        _ = client.CreateTransfers(new[] { transfer });
         Assert.Fail();
     }
 
 
-    private static void AssertAccounts(Account[] lookupAccounts)
+    private static void AssertAccounts(Account[] expected, Account[] actual)
```
Comment (matklad) [inline src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:1520]: Can we use this thing here? 

https://stackoverflow.com/a/11055686

hopefully it’ll print readable diff on mismatxhn

File: src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs (hunk 1 above)
Comment (batiati) [reply src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs]: It's due to the compiler null checks. I'd rather use those `Assert.IsNotNull(server)` than the null-forgiving syntax `server!` everywhere.

File: src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs (hunk 2 above)
Comment (batiati) [reply src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:1520]: We can't compare all fields; for example, the balances can change between creation and lookup.

File: src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs (hunk 1 above)
Comment (matklad) [reply src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs]: Ahh, that makes sense. Seems like C# has an escape hatch here?

https://stackoverflow.com/a/73167373

Could we declare the relevant fields as non-null and init then with `null!`?

**Review by matklad — APPROVED** [review]

**Pull request merged**